axelrod/ultimatum/player.py

```diff
@@ -12,8 +14,12 @@ class PlayerPosition(Enum):
     DECIDER = 2
 
 
-# Python 3.7 dataclass would be nice here
-Outcome = namedtuple('Outcome', ['position', 'offer', 'decision', 'scores'])
+@attr.s
+class Outcome(object):
+    position: "UltimatumPlayer" = attr.ib()
```

Comment:
What's going on here with "UltimatumPlayer"? Is it a generic default?

Reply:
I'm doing a forward reference to the type that we will declare later.

https://www.python.org/dev/peps/pep-0484/#forward-references